kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/kommunicate/views/KmAssigneeListFragment.java

```diff
@@ -0,0 +1,171 @@
+import com.applozic.mobicomkit.uiwidgets.conversation.adapter.KmUserPagerAdapter;
+import com.applozic.mobicomkit.uiwidgets.conversation.fragment.KmUserListFragment;
+import com.applozic.mobicomkit.uiwidgets.databinding.KmAssigneeListLayoutBinding;
+import com.applozic.mobicommons.commons.core.utils.Utils;
+import com.applozic.mobicommons.json.GsonUtils;
+import com.applozic.mobicommons.people.channel.Channel;
+import com.google.android.material.tabs.TabLayout;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.kommunicate.Kommunicate;
+import io.kommunicate.callbacks.KMGetContactsHandler;
+import io.kommunicate.users.KmAssigneeListHelper;
+import io.kommunicate.users.KmContact;
+
+public class KmAssigneeListFragment extends Fragment implements TabLayout.OnTabSelectedListener, SearchView.OnQueryTextListener {
+    private static final String TAG = "KmAssigneeListFragment";
+    private String assigneeId;
+    private KmAssigneeListLayoutBinding binding;
+    private KmUserPagerAdapter pagerAdapter;
+    private String searchText;
+    private int channelId;
+
+    public static String getFragTag() {
+        return TAG;
+    }
+
+    public static KmAssigneeListFragment newInstance(String assigneeId, int channelId) {
+        KmAssigneeListFragment assigneeListFragment = new KmAssigneeListFragment();
+        Bundle args = new Bundle();
+        args.putString(Channel.CONVERSATION_ASSIGNEE, assigneeId);
+        args.putInt(ConversationUIService.GROUP_ID, channelId);
+        assigneeListFragment.setArguments(args);
+        return assigneeListFragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Bundle bundle = getArguments();
+        if (bundle != null) {
+            assigneeId = bundle.getString(Channel.CONVERSATION_ASSIGNEE, null);
+            channelId = bundle.getInt(ConversationUIService.GROUP_ID);
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.km_assignee_list_layout, container, false);
+
+        binding = DataBindingUtil.bind(view.findViewById(R.id.kmAssigneeListLayout));
+        List<String> titleList = new ArrayList<>();
+        titleList.add(getString(R.string.km_agent_list_title));
+        titleList.add(getString(R.string.km_bot_list_title));
+
+        if (binding != null) {
+            binding.setAssigneeListFragment(this);
+            pagerAdapter = new KmUserPagerAdapter(getFragmentManager(), titleList);
+            KmUserListFragment agentsFragment = KmUserListFragment.getInstance(assigneeId, true, channelId, KmAssigneeListHelper.getAgentList());
+            KmUserListFragment botsFragment = KmUserListFragment.getInstance(assigneeId, false, channelId, KmAssigneeListHelper.getBotList());
```

Comment:
is KmUserListFragment a singleton? if yes, then above thing might not be correct because both agents and bots will have a different object. if no, then need to correct the naming

Reply:
Its not singleton, the getInstance method just returns the fragment object by setting arguments to it. Otherwise we need to write this code when initialising the fragment.

Reply:
ok then we can change the method naming to make it more understandable.

Reply:
done